src/ServicePulse.Host/vue/src/components/failedmessages/AllDeletedMessages.vue

```diff
@@ -57,18 +61,13 @@ function loadMessages() {
   return loadPagedMessages(groupId.value, pageNumber.value, "", "", startDate.toISOString(), endDate.toISOString());
 }
 
-async function loadGroupDetails(groupId) {
+async function loadGroupDetails(groupId: string | string[]) {
   const response = await useFetchFromServiceControl(`archive/groups/id/${groupId}`);
   const data = await response.json();
   groupName.value = data.title;
 }
 
-function loadPagedMessages(groupId, page, sortBy, direction, startDate, endDate) {
-  if (typeof sortBy === "undefined") sortBy = "modified";
-  if (typeof direction === "undefined") direction = "desc";
-  if (typeof page === "undefined") page = 1;
-  if (typeof startDate === "undefined") startDate = new Date(0).toISOString();
-  if (typeof endDate === "undefined") endDate = new Date().toISOString();
+function loadPagedMessages(groupId?: string | string[], page: number = 1, sortBy: string = "modified", direction: string = "desc", startDate: string = new Date(0).toISOString(), endDate: string = new Date().toISOString()) {
```

Comment:
`groupId?: string | string[]` feels wrong to me given we are doing string concatenation.
`sortBy: string = "modified"` should these be keyed to the allowed types, I can't actually see a `modified` prop in `FailedMessageViewWithExtendedUIProperties`. I would recommend using types for these, eg `sortBy: keyof FailedMessageViewWithExtendedUIProperties = "last_modified",`, but I would checkin ServiceControl to see what sort columns are allowed.
Same for `direction`, this should be an enum.